cdi/src/test/java/io/smallrye/config/inject/ValidateInjectionTest.java

```diff
@@ -26,68 +28,214 @@
+ * </pre>
+ * 
+ * where each Exception is a supressedException.
+ */
 public class ValidateInjectionTest {
+
     @Test
-    void missingProperty() {
-        JupiterTestEngine engine = new JupiterTestEngine();
-        LauncherDiscoveryRequest request = request().selectors(selectClass(MissingPropertyTest.class)).build();
-        EngineExecutionResults results = EngineTestKit.execute(engine, request);
-        results.testEvents().failed().assertEventsMatchExactly(finishedWithFailure(instanceOf(DeploymentException.class),
-                message("SRCFG02000: No Config Value exists for required property missing.property")));
+    void missingProperty() throws Exception {
+        DeploymentException exception = getDeploymentException(MissingPropertyTest.class);
+        assertThat(exception).hasMessage(
+                "SRCFG02000: Failed to Inject @ConfigProperty for key missing.property into missingProp since the config property could not be found in any config source");
+
+        assertThat(exception.getCause()).isInstanceOf(ConfigInjectionException.class);
+        assertThat(exception.getCause()).hasMessage(
+                "SRCFG02000: Failed to Inject @ConfigProperty for key missing.property into missingProp since the config property could not be found in any config source");
     }
 
     @Test
-    void converterMissingProperty() {
-        JupiterTestEngine engine = new JupiterTestEngine();
-        LauncherDiscoveryRequest request = request().selectors(selectClass(ConverterMissingPropertyTest.class)).build();
-        EngineExecutionResults results = EngineTestKit.execute(engine, request);
-        results.testEvents().failed().assertEventsMatchExactly(finishedWithFailure(instanceOf(DeploymentException.class),
-                message("SRCFG02000: No Config Value exists for required property missing.property")));
+    void emptyProperty() throws Exception {
+        DeploymentException exception = getDeploymentException(EmptyPropertyTest.class);
+        assertThat(exception)
+                .hasMessageStartingWith(
+                        "SRCFG02001: Failed to Inject @ConfigProperty for key empty.property into emptyProp SRCFG00040:");
+
+        assertThat(exception.getCause()).isInstanceOf(ConfigInjectionException.class);
+
+        assertThat(exception.getCause().getCause()).isInstanceOf(NoSuchElementException.class);
+        assertThat(exception.getCause().getCause()).hasMessage(
+                "SRCFG00040: The config property empty.property is defined as the empty String (\"\") which the following Converter considered to be null: io.smallrye.config.Converters$BuiltInConverter");
+    }
+
+    @Test
+    void badProperty() throws Exception {
+        DeploymentException exception = getDeploymentException(BadPropertyTest.class);
+        assertThat(exception)
+                .hasMessageStartingWith(
+                        "SRCFG02001: Failed to Inject @ConfigProperty for key bad.property into badProp SRCFG00041:");
```

Comment:
I don't think this is enough information for the user to identify the injection location.

I would use the following to identify the location:
* For a field: `com.example.MyClass.myField`
* For a method parameter: `com.example.MyClass.myMethod(String, int, AnotherClass)`
* For a constructor parameter: `com.example.MyClass(String, int, AnotherClass)`

This might require either a helper method to do the conversion from a `Member` to a string, or a helper class and  [`@FormatWith`](https://jboss-logging.github.io/jboss-logging-tools/apidocs/org/jboss/logging/annotations/FormatWith.html)

Reply:
+1

Reply:
I'm going to give this a try hopefully some time next week. Thanks :)